### PR TITLE
Additional Checker Method

### PR DESCRIPTION
**Pull request by saquino0827**
### Information
- QPPCT-683

### Changes proposed in this PR:
- Adds a new `childExact` checker method with tests
- Refactor childMin/childMax combo. with childExact.
- Refactor/remove error messaging.

### Checklist 
- [x] All JUnit tests pass (`mvn clean verify`).
- [x] New unit tests written to cover new functionality.
- [x] Added and updated JavaDocs for non-test classes and methods.
- [x] No local design debt. Do you feel that something is "ugly" after your changes?
- [n/a] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

File: converter/src/test/java/gov/cms/qpp/conversion/validate/CheckerTest.java [hunk 1]
```diff
@@ -169,6 +169,46 @@ void testChildrenMaximumSuccess() {
...
+		Node meepNode = new Node();
+		meepNode.addChildNodes(new Node(TemplateId.PLACEHOLDER),
+			new Node(TemplateId.ACI_AGGREGATE_COUNT),
+			new Node(TemplateId.PLACEHOLDER));
+
+		Checker checker = Checker.check(meepNode, details);
+		checker.childExact(ERROR_MESSAGE, 2, TemplateId.PLACEHOLDER, TemplateId.ACI_AGGREGATE_COUNT);
+
+		assertWithMessage("An error exists")
+			.that(details).hasSize(1);
```
Comment (halprin) [inline converter/src/test/java/gov/cms/qpp/conversion/validate/CheckerTest.java]: We should test that we get the expected error and not just the correct number of errors.

**Comment by sfradkin** [comment]
Are there any opportunities to do some refactoring to pull up common checks?  There appear to be a lot of node types for which we do the same or very similar checks.

**Comment by saquino0827** [comment]
@sfradkin I am currently trying to find and remove common checks. I assume this is what you meant.

@halprin made the appropriate change from your suggestion and looking at more validations for the similar checks.

**Comment by sfradkin** [comment]
@saquino0827: I would prefer a way to remove the duplication of code if possible.  There may not be any way to do this because we don't have typed node classes, they're all generic.

**Comment by saquino0827** [comment]
@sfradkin I think I got an idea of removing the extra exactly on child errors. I can refactor the method to check all types passed in and format an error message appropriate for each that is invalid. It will at least remove duplicate childExact calls for each template id. There are some common checks I cannot avoid, but I'll do my best to see what code duplication I can remove. There may be more checker methods that could be refactored so that we don't call them multiple times for different template id's as well.

**Comment by saquino0827** [comment]
@sfradkin I decided against refactoring the checker method for childExact. Although we obtain the ability to call the method with different template ids, the error message handling for generic types and specific types becomes more convoluted. There was no clear way to refactor the message handling therefore I felt more inclined to keep the code the way it is now.

File: converter/src/main/java/gov/cms/qpp/conversion/validate/AciSectionValidator.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import gov.cms.qpp.conversion.model.Node;
 import gov.cms.qpp.conversion.model.TemplateId;
 import gov.cms.qpp.conversion.model.Validator;
+import gov.cms.qpp.conversion.model.error.Error;
```
Comment (FlexionSonarQube) [inline converter/src/main/java/gov/cms/qpp/conversion/validate/AciSectionValidator.java]: MINOR Remove this unused import 'gov.cms.qpp.conversion.model.error.Error'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by FlexionSonarQube** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by FlexionSonarQube** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AciSectionValidator.java#L6](https://github.com/CMSgov/qpp-conversion-tool/blob/618cbc5187ab461ed3c82ea24725278a9f60bb02/converter/src/main/java/gov/cms/qpp/conversion/validate/AciSectionValidator.java#L6): Remove this unused import 'gov.cms.qpp.conversion.model.error.Error'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: ERROR_MESSAGES.md [hunk 3]
```diff
@@ -43,8 +43,7 @@ Any text in the following format `(Example)` are considered variables to be fill
 * 36 : CT - Must contain one Measure (eCQM) section
 * 37 : CT - Must contain correct number of performance rate(s). Correct Number is `(Expected value)`
 * 38 : CT - This `(Numerator or Denominator)` Node does not have any child Nodes
-* 39 : CT - This `(Numerator or Denominator)` Node does not have an Aggregate Count Node
-* 40 : CT - This `(Numerator or Denominator)` Node has too many child Nodes
+* 39 : CT - This `(Numerator or Denominator)` Node must have exactly one Aggregate Count node
```
Comment (clydet) [inline ERROR_MESSAGES.md]: And now... there IS no ErrorCode 40. Cue the spooky campfire story.

File: ERROR_MESSAGES.md [hunk 4]
```diff
@@ -12,16 +12,16 @@ Any text in the following format `(Example)` are considered variables to be fill
 * 7 : CT - The measure reference results must have at least one measure
 * 8 : CT - A single aggregate count value is required
 * 9 : CT - Aggregate count value must be an integer
```
Comment (clydet) [inline ERROR_MESSAGES.md]: Worth noting that this file is auto generated from ErrorCode.java. If you run `mvn clean package` that should regenerate this file with what ever updates you've made to ErrorCode.

File: commons/src/main/java/gov/cms/qpp/conversion/model/error/ErrorCode.java [hunk 5]
```diff
@@ -31,11 +31,9 @@
 	CHILD_MEASURE_MISSING(7, "The measure reference results must have at least one measure"),
 	AGGREGATE_COUNT_VALUE_NOT_SINGULAR(8, "A single aggregate count value is required"),
 	AGGREGATE_COUNT_VALUE_NOT_INTEGER(9, "Aggregate count value must be an integer"),
-	ACI_MEASURE_PERFORMED_RNR_MEASURE_PERFORMED_MISSING(10, "ACI Measure Performed RnR's Measure Performed is "
-			+ "required"),
-	ACI_MEASURE_PERFORMED_RNR_MEASURE_PERFORMED_REPEATED(11, "ACI Measure Performed RnR's Measure Performed "
-			+ "can only be present once"),
-	ACI_MEASURE_PERFORMED_RNR_MEASURE_ID_NOT_SINGULAR(12, "ACI Measure Performed RnR's requires a single "
+	ACI_MEASURE_PERFORMED_RNR_MEASURE_PERFORMED_EXACT(11, "This ACI Measure Performed RnR requires exactly one "
```
Comment (clydet) [inline commons/src/main/java/gov/cms/qpp/conversion/model/error/ErrorCode.java]: No more 10?

Comment (saquino0827) [reply commons/src/main/java/gov/cms/qpp/conversion/model/error/ErrorCode.java]: Ah yes. I need to run an update on the file. Good catch.

File: ERROR_MESSAGES.md (hunk 3 above)
Comment (saquino0827) [reply ERROR_MESSAGES.md]: Are you afraid of the dark?

File: ERROR_MESSAGES.md (hunk 4 above)
Comment (saquino0827) [reply ERROR_MESSAGES.md]: Yeah.. some of the validations were duplicates of other checks. When I removed the validations, the messages were no longer needed. I will run another `mvn clean install` to ensure my error messages are up to date.

File: commons/src/main/java/gov/cms/qpp/conversion/model/error/ErrorCode.java (hunk 5 above)
Comment (saquino0827) [reply commons/src/main/java/gov/cms/qpp/conversion/model/error/ErrorCode.java]: This was one of those conditions where we check if the child exists and then check if it has exactly one of the child needed. A redundant check.

**Pull request merged**